core/src/main/java/org/mule/expression/ExpressionUtils.java

```diff
@@ -281,6 +282,48 @@ public static TypedValue getTypedProperty(String expression, MuleMessage msg, Pr
         return new TypedValue(result, propertyDataType);
     }
 
+    /**
+     * Returns {@code true} if the {@code expression} string starts with
+     * the {@code parser}' style prefix and ends with the style's suffix.
+     * Notice that this doesn't account for the case in which you have an embedded
+     * expression like {@code hello #[world]}
+     *
+     * @param value  a {@link String} which is presumed to be a simple expression
+     * @param parser the {@link TemplateParser} which provides the {@link TemplateParser.PatternInfo} style
+     * @return {@code true} if {@code expression} is a simple expression. {@code false otherwise}
+     */
+    public static boolean isSimpleExpression(String value, TemplateParser parser)
+    {
+        TemplateParser.PatternInfo style = parser.getStyle();
+        return value.startsWith(style.getPrefix()) && value.endsWith(style.getSuffix());
```

Comment:
it's a simple expression in the sense that it's a whole expression and not an embedded one